docs/patterns/_code-snippets.md

```diff
@@ -0,0 +1,35 @@
+---
+collection: patterns
+title: code snippets
+---
+
+## .code-snippet
+
+Code snippets should be used when presenting the user with a small snippet of code that they will likely want to copy and paste.
+
+```html
+<div class="code-snippet">
+    <input class="code-snippet__input" value="sudo apt-get update" readonly="readonly">
+    <button class="code-snippet__copy-button">Copy to clipboard</button>
+</div>
+```
+
+<div class="code-snippet">
+    <input class="code-snippet__input" value="sudo apt-get update" readonly="readonly">
+    <button class="code-snippet__action">Copy to clipboard</button>
+</div>
+
+## .code-snippet--inverse
+
+
+```html
+<div class="code-snippet code-snippet--inverse">
```

Comment:
This is the original strict BEM method. 

Digging out your BEM guidance doc from a few months back, I see a discussion on this but no conclusions? In general, I favour explicit naming for greater transparency - it's immediately clear from the markup what classes this element is using.

I try to avoid @extend where possible as it's very easy to muddle specificity although this is perhaps one case where it's justified given each modifier has a direct and close relationship to it's parent.